flang/lib/Optimizer/Builder/FIRBuilder.cpp

```diff
@@ -1759,3 +1759,38 @@ fir::factory::deduceOptimalExtents(mlir::ValueRange extents1,
   }
   return extents;
 }
+
+llvm::SmallVector<mlir::Value> fir::factory::updateRuntimeExtentsForEmptyArrays(
+    fir::FirOpBuilder &builder, mlir::Location loc, mlir::ValueRange extents) {
+  if (extents.size() <= 1)
+    return extents;
+
+  // Try to reduce the number of new zero constant operations.
+  // This just makes MLIR matching easier, if CSE is not run
+  // after this.
```

Comment:
Which matching do you have in mind? I'm not against (it is easier to read for humans and this isn't too heavy weight), but this isn't the normal way I've seen things work.

I'm not sure how much this helps matching because there is no guarantee there won't already be other zeros in the same function.

Reply:
Addressing Jean's question as well.

Yes, it is not usual for extents to have different data types, because they will likely be casted to `index` somewhere else (e.g. as in elemental codegen, where we use `getIndexExtents`).

I guess I overthought it, and I will remove this code :)